CONTRIBUTING.markdown

```diff
@@ -1,106 +1,124 @@
+* Browse through [the Jekyll discussion forum](https://talk.jekyllrb.com/), and lend a hand answering questions. There's a good chance you've already experienced what another user is experiencing.
+* Find [an open issue](https://github.com/jekyll/jekyll/issues) (especially [those labeled `help-wanted`](https://github.com/jekyll/jekyll/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted)), and submit a proposed fix. If it's your first pull request, we promise we won't bite, and are glad to answer any questions.
+* Help evaluate [open pull requests](https://github.com/jekyll/jekyll/pulls), by testing the changes locally and reviewing what's proposed.
 
-Both `script/test` and `script/cucumber` can be run without arguments to
-run its entire respective suite.
+## Submitting a pull request
+
+### Pull requests generally
+
+* The smaller the proposed change, the better. If you'd like to propose two unrelated changes, submit two pull requests.
+
+* The more information, the better. Make judicious use of the pull request body. Describe what changes were made, why you made them, and what impact they will have for users.
+
+* If you're submitting a code contribution, be sure to read the [code contributions](#code-contributions) section below.
+
+* Don't bump the Gem version in your pull request (if you don't know what that means, you probably didn't).
+
+### Submitting a pull request via github.com
+
+Many small changes can be made entirely through the github.com web interface.
+
+1. Navigate to the file within [`jekyll/jekyll`](https://github.com/jekyll/jekyll) that you'd like to edit
+2. Click the pencil icon in the top right corner to edit the file
+3. Make your proposed changes
+4. Click "Propose file change"
+5. Click "Create pull request"
+6. Add a descriptive title and detailed description for your proposed change. The more information the better.
+7. Click "Create pull request"
+
+That's it! You'll be automatically subscribed to receive updates as others review your proposed change and provide feedback.
+
+### Submitting a pull request via Git command line
+
+1. Fork the project by clicking "Fork" in the top right corner of [`jekyll/jekyll`](https://github.com/jekyll/jekyll)
+2. Clone the repository lcoally `git clone https://github.com/<you-username>/jekyll`
+3. Create a new, descriptively named branch to contain your change ( `git checkout -b my-awesome-feature` ).
+4. Hack away, add tests. Not necessarily in that order.
+5. Make sure everything still passes by running `script/cibuild` (see [the tests section](#running-tests-locally) below)
+6. Push the branch up ( `git push origin my-awesome-feature` ).
+7. Create a pull request by visiting https://github.com/<your-username>/jekyll/ and following the instructions at the top of the screen.
+
+## Proposing updates to the documentation
+
+We want the Jekyll documentation to be the best it can be. We've open-sourced our docs and we welcome any pull requests if you find it lacking.
```

Comment:
`open-sourced our docs` could link to github.com/jekyll/jekyll/tree/master/site?